_studio/mfx_lib/encode_hw/mpeg2/src/mfx_mpeg2_encode_utils_hw.cpp

```diff
@@ -1309,6 +1309,16 @@ namespace MPEG2EncoderHW
             }
         }
 
+        if (m_VideoParamsEx.mfxVideoParams.mfx.RateControlMethod == MFX_RATECONTROL_CQP)
+        {
+            if (m_VideoParamsEx.mfxVideoParams.mfx.BufferSizeInKB != 0)
```

Comment:
@ogolikov, @splotnikv , @lakulako , can you please review this particular part ? this code block is essential. I mean is it okay to turn on CQP_HRD mode if  a user provides non-zero BufferSizeInKB or we need to add other conditions here?

Reply:
What is the new mode? Does it requires bitrate to be provided?